packages/core/src/extension/extension-action-creator.ts

```diff
@@ -34,4 +37,37 @@ export class ExtensionActionCreator {
                     });
             });
     }
+
+    renderExtension(
+        container: string,
+        region: ExtensionRegion,
+    ): ThunkAction<ExtensionAction, InternalCheckoutSelectors> {
+        return (store) =>
+            Observable.create((observer: Observer<ExtensionAction>) => {
+                const state = store.getState();
+                const { id: cartId } = state.cart.getCartOrThrow();
+                const extension = state.extensions.getExtensionByRegion(region);
+
+                if (!extension) {
+                    throw new ExtensionNotFoundError(
+                        `Unable to proceed due to no extension configured for ${region}.`,
+                    );
+                }
+
+                observer.next(createAction(ExtensionActionType.RenderExtensionRequested));
+
+                try {
+                    const iframe = new ExtensionIframe(container, extension, cartId);
+
+                    iframe.attach();
+
+                    observer.next(createAction(ExtensionActionType.RenderExtensionSucceeded));
```

Comment:
Do we want to resolve the async action only when the iframe is rendered successfully? At the moment, it is resolved as soon as the iframe is attached, meaning that it might not be actually rendered.

Reply:
Thanks DC for your review.
`iframe.attach();` is still WIP. In the next PR, by the time `iframe.attach()` finishes, the iframe should have adjusted its own size based on the iframe content.